cli/internal/packager/common.go

```diff
@@ -134,50 +152,49 @@ func HandleIfURL(packagePath string, shasum string, insecureDeploy bool) string
 	// Check if the user gave us a remote package
 	providedURL, err := url.Parse(packagePath)
 	if err != nil || providedURL.Scheme == "" || providedURL.Host == "" {
-		logrus.WithField("archive", packagePath).Debug("The package provided is not a remote package.")
 		return packagePath
 	}
 
 	if !insecureDeploy && shasum == "" {
-		logrus.Fatal("When deploying a remote package you must provide either a `--shasum` or the `--insecure` flag. Neither were provided.")
+		message.Fatal(nil, "When deploying a remote package you must provide either a `--shasum` or the `--insecure` flag. Neither were provided.")
 	}
 
 	// Check the extension on the package is what we expect
 	if !isValidFileExtension(providedURL.Path) {
-		logrus.Fatalf("Only %s file extensions are permitted.\n", config.GetValidPackageExtensions())
+		message.Fatalf(nil, "Only %s file extensions are permitted.\n", config.GetValidPackageExtensions())
 	}
 
 	// Download the package
 	resp, err := http.Get(packagePath)
 	if err != nil {
-		logrus.Fatal("Unable to download the package: ", err)
+		message.Fatal(err, "Unable to download the package")
 	}
 	defer resp.Body.Close()
 
 	// Write the package to a local file
 	tempPath := createPaths()
+	defer tempPath.clean()
```

Comment:
I think we need to think of a better way to do this, either pass off the responsibility of clearing to somewhere else or return the package body instead of a path to the local filesystem.

This defer wipes away the downloaded package before it's able to actually be used during the `deploy` stage.

Reply:
Or even simpler.. We can create the tmp directory in the `Run` section of the cobra DeployCmd and leave the defer in there since the first thing `packager.Deploy()` does is create the /tmp/ directory anyways.

Reply:
the way we do it everywhere in the codebase is to pass the parent temp path and just let it be handled by the parent deploy function